src/graphql/resolvers/index.js

```diff
@@ -168,49 +169,85 @@ module.exports = function Resolvers (ssb) {
   }
 
   /**
-   * Takes the messages from a thread and maps them based on whether
-   * a profile is returned for the author of that message or not.
-   * Messages from someone who hasnt yet opted in to publicWebHosting or
-   * a profile wasnt found for them, will return empty values
-   * @param {array} messages - messages in a thread
+   * Takes a messages and maps it either to a Comment or "empty" Comment depending on
+   * whether the author has opted into publicWebHosting.
+   * @param {object} msg - a post message in kv format
+   * @param {function} cb - an error first callback function
    */
-  const mapMessages = (messages) => {
-    return new Promise((resolve, reject) => {
-      pull(
-        pull.values(messages),
-        pullParaMap((msg, cb) => {
-          getProfile(msg.value.author)
-            .then(profile => {
-              // if there was a profile, thats great
-              if (profile) {
-                return cb(null, {
-                  id: msg.key,
-                  author: msg.value.author,
-                  text: msg.value.content.text,
-                  timestamp: msg.value.timestamp // asserted publish time
-                })
+  function publicifyMessage (msg, cb) {
+    const message = {
+      id: null,
+      author: null,
+      timestamp: msg.value.timestamp, // asserted publish time
+      text: null
+    }
+
+    canPubliclyHost(msg.value.author)
+      .then(canHost => {
+        // if there was a profile, that means publicWebHosting === true
+        if (canHost) {
+          message.id = msg.key
+          message.author = msg.value.author
+          message.text = msg.value.content.text
+        }
+
+        cb(null, message)
+      })
+      .catch(_err => cb(null, message))
+  }
+
+  const roomState = {
+    name: null,
+    members: new Set()
+  }
```

Comment:
I don't think this file is the right place to be storing state and updating it... but it will do for now